src/nupic/algorithms/Connections.cpp

```diff
@@ -473,6 +473,7 @@ void Connections::raisePermanencesToThreshold(
     return;
 
   vector<Synapse> &synapses = segData.synapses;
+  if(synapses.size() <= 0) return; 
```

Comment:
@ctrl-z-9000-times this workarounds a bug, where with some settings, SP would produce a segment with empty synapses, and Conn::raisePermanenceToThreshold would crash with it! 

This return avoids the crash. 
But is it a proper solution, or a segment with empty synapses should not happen in the first place?

Reply:
resolved in #246

Reply:
Ok, I had to merge to work on this branch.  After I push my changes you need to check that I did the conflict resolution in Connections.cpp around 476 correctly.